pkg/syncer/storagepool/service.go

```diff
@@ -132,26 +132,15 @@ func InitStoragePoolService(ctx context.Context,
 	go func() {
 		diskDecommEnablementTicker := time.NewTicker(common.DefaultFeatureEnablementCheckInterval)
 		defer diskDecommEnablementTicker.Stop()
-		clusterFlavor := cnstypes.CnsClusterFlavorWorkload
 		for ; true; <-diskDecommEnablementTicker.C {
-			coCommonInterface, err := commonco.GetContainerOrchestratorInterface(ctx,
-				common.Kubernetes, clusterFlavor, *coInitParams)
+			_, err := initDiskDecommController(ctx, migrationController)
 			if err != nil {
-				log.Errorf("Failed to create CO agnostic interface. Error: %v", err)
+				log.Warnf("Error while initializing disk decommission controller. Error: %+v. "+
+					"Retry will be triggered at %v",
+					err, time.Now().Add(common.DefaultFeatureEnablementCheckInterval))
 				continue
 			}
-			if !coCommonInterface.IsFSSEnabled(ctx, common.VSANDirectDiskDecommission) {
-				log.Infof("VSANDirectDiskDecommission feature is disabled on the cluster")
-			} else {
-				_, err := initDiskDecommController(ctx, migrationController)
```

Comment:
we are not calling `initDiskDecommController` because it is already initialized before since the feature is enabled. Can you confirm?

Reply:
I see we are calling at line 136